components/public-api/gitpod/experimental/v1/teams.proto

```diff
@@ -73,6 +84,9 @@ service TeamsService {
     // ListTeams lists the caller has access to.
     rpc ListTeams(ListTeamsRequest) returns (ListTeamsResponse) {};
 
+    // GetTeamList lists teams that the caller has access to.
+    rpc GetTeamList(GetTeamListRequest) returns (GetTeamListResponse) {};
```

Comment:
> It is not consistent then with other functions. Maybe ListTeams2?

Sounds good

> What was wrong with making properties optional in ListTeamsResponse?

@akosyakov  The actual improvement we want is to remove respond of `listTeam`  `members` property, so we need to update public api server. Once it's been removed, dashboard with old version will be broken.

In this case, deprecate or optional is the same. No compatibility if upgrade from old version

So if we thought this page/APIs are important and need more compatibility, then we should never deprecate related proto, But add new API and deprecate old API after a long time like x months

------

But I also ask questions for compatibility rollout [[slack](https://gitpod.slack.com/archives/C05H5UQBW6Q/p1697543896360139?thread_ts=1697543877.891219&cid=C05H5UQBW6Q)]. If we add compatibility code, it should be there for x months, I don't like it.

And we also want to introduce real PAPI, then the compatibility code will be 🤯 

------

Or you mean that, check to code to see if optional members and invitation not affect anything? We check isOwner from members, so it can affect lots of places

Reply:
We chatted today with Milan/Chris/Sven and outcome was to ensure order of component deployments in Dedicated and Cloud. Maybe it is what we should look next instead.

And particularly this time we will simply wait for complete rollout? let's chat tomorrow?